### PR TITLE
Adding highlight support in the search

### PR DESCRIPTION
**Pull request by arpit-at**
[Highlight](https://www.elastic.co/guide/en/elasticsearch/reference/current/highlighting.html) support is added 

 Changes in request 

1. `highlight` needs to be added as the top-level attribute in the DSL
```
 "highlight": {
            "fields": {
                "name": {},
                "displayName": {},
                "description": {},
                "userDescription": {},
                "__meaningsText": {},
                "__meaningNames.text": {}
            }
        }
```
2. `showHighlights`  needs to be set to true if it is required in response
 Changes in Response 

A new object in response will come when `showHighlights` is set to true (default is false)
```
 "searchMetadata": {
        "810b0998-27c0-4feb-9606-b2863c4a2651": {
            "highlights": {
                "name": [
                    "<em>arpit</em>_6"
                ],
                "description": [
                    "<em>arpit</em>_6"
                ],
                "userDescription": [
                    "<em>arpit</em>_6"
                ],
                "displayName": [
                    "<em>arpit</em>_6"
                ]
            }
        },
        "3632ebdd-e7cd-4aca-bca9-0ee91c6445db": {
            "highlights": {
                "name": [
                    "<em>arpit</em>_5"
                ],
                "description": [
                    "<em>arpit</em>_5"
                ],
                "userDescription": [
                    "<em>arpit</em>_5"
                ],
                "displayName": [
                    "<em>arpit</em>_5"
                ]
            }
        }
```

- [x] tested on local

### CODE REVIEW

File: graphdb/api/src/main/java/org/apache/atlas/repository/graphdb/DirectIndexQueryResult.java [hunk 1]
```diff
@@ -8,6 +8,8 @@
 public class DirectIndexQueryResult<V, E> {
     private Iterator<AtlasIndexQuery.Result<V, E>> iterator;
     private Map<String, Aggregation> aggregationMap;
+
+    private Map<String, Object> highlightMap;
```
Comment (mehtaanshul) [inline graphdb/api/src/main/java/org/apache/atlas/repository/graphdb/DirectIndexQueryResult.java:12]: Don't see this getting used anywhere

File: graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java [hunk 2]
```diff
@@ -336,5 +341,10 @@ public double getScore() {
             }
             return Double.parseDouble(String.valueOf(hit.get("_score")));
         }
+
+        @Override
+        public Map<String, List<String>> getHighLights() {
+            return (Map<String, List<String>>) this.hit.get("highlight");
```
Comment (mehtaanshul) [inline graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java:347]: Add null check here. this.hit.get("highlight") might return null

File: intg/src/main/java/org/apache/atlas/model/discovery/AtlasSearchResult.java [hunk 3]
```diff
@@ -58,6 +59,10 @@ public class AtlasSearchResult implements Serializable {
     private Map<String, Object>            aggregations;
     private Map<String,Double>             searchScore;
 
+    private Map<String, SearchMetadata>   searchMetadata;
```
Comment (mehtaanshul) [inline intg/src/main/java/org/apache/atlas/model/discovery/AtlasSearchResult.java:62]: This is becoming a little confusing, SearchMetadata is Map of SearchMetadata. We can do better naming here.

File: graphdb/api/src/main/java/org/apache/atlas/repository/graphdb/DirectIndexQueryResult.java (hunk 1 above)
Comment (arpit-at) [reply graphdb/api/src/main/java/org/apache/atlas/repository/graphdb/DirectIndexQueryResult.java]: removed

File: graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java (hunk 2 above)
Comment (arpit-at) [reply graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java]: added

File: intg/src/main/java/org/apache/atlas/model/discovery/AtlasSearchResult.java (hunk 3 above)
Comment (arpit-at) [reply intg/src/main/java/org/apache/atlas/model/discovery/AtlasSearchResult.java]: refactored

File: graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java [hunk 4]
```diff
@@ -280,6 +280,11 @@ public Set<String> getCollapseKeys() {
         public DirectIndexQueryResult<AtlasJanusVertex, AtlasJanusEdge> getCollapseVertices(String key) {
             return null;
         }
+
+        @Override
+        public Map<String, List<String>> getHighLights() {
+            return null;
```
Comment (n5nk) [inline graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java:286]: nit: donot return null. Return an empty map instead

File: graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java [hunk 5]
```diff
@@ -336,5 +341,14 @@ public double getScore() {
...
             return Double.parseDouble(String.valueOf(hit.get("_score")));
         }
+
+        @Override
+        public Map<String, List<String>> getHighLights() {
+            Object highlight = this.hit.get("highlight");
+            if(Objects.isNull(highlight)) {
+                return (Map<String, List<String>>) highlight;
+            }
+            return null;
```
Comment (n5nk) [inline graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java:351]: same here, return empty map

File: intg/src/main/java/org/apache/atlas/model/discovery/ElasticsearchMetadata.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    public Map<String, List<String>> getHighlights() {
+        return highlights;
+    }
+
+    public void setHighlights(Map<String, List<String>> highlights) {
+        this.highlights = highlights;
+    }
+
+    public void addHighlights(Map<String, List<String>> highlights) {
```
Comment (n5nk) [inline intg/src/main/java/org/apache/atlas/model/discovery/ElasticsearchMetadata.java:21]: `setHighlights` and `addHighlights` seems redundant. Please use `addHighlights` only

File: graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java (hunk 5 above)
Comment (arpit-at) [reply graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java]: changed

File: intg/src/main/java/org/apache/atlas/model/discovery/ElasticsearchMetadata.java (hunk 6 above)
Comment (arpit-at) [reply intg/src/main/java/org/apache/atlas/model/discovery/ElasticsearchMetadata.java:17]: `setHighlights ` removed

**Review by n5nk — APPROVED** [review]

**Pull request merged**